### PR TITLE
fix(storefront): BCTHEME-1781 ADA Compliance - The Add to Cart modal dialog container is not labeled

### PR DESCRIPTION
**Pull request by bc-yevhenii-buliuk**
#### What?

This PR adds the appropriate aria attributes to Add to Cart modal dialog container.

#### Tickets / Documentation

- [BCTHEME-1781](https://bigcommercecloud.atlassian.net/browse/BCTHEME-1781)
- [Dialog (Modal) Pattern](https://www.w3.org/WAI/ARIA/apg/patterns/dialog-modal)

#### Screen Recording
**before changes:**

https://github.com/bigcommerce/cornerstone/assets/82589781/bdbe961a-df5e-43ec-8c98-bd811d6523ab

**after changes:**

https://github.com/bigcommerce/cornerstone/assets/82589781/df160a86-60e9-4a65-af61-638fecbda027

[BCTHEME-1781]: https://bigcommercecloud.atlassian.net/browse/BCTHEME-1781?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: templates/components/products/product-view.html [hunk 1]
```diff
@@ -274,7 +274,14 @@ <h2 class="productView-brand">
     </article>
 </div>
 
-<div id="previewModal" class="modal modal--large" data-reveal>
+<div
+    id="previewModal"
+    class="modal modal--large"
+    role="alertdialog"
```
Comment (yurytut1993) [inline templates/components/products/product-view.html:280]: `alertdialog` role should only used when an alert, error, or warning occurs. Can we change it to `dialog`?

Comment (bc-yevhenii-buliuk) [reply templates/components/products/product-view.html:280]: @yurytut1993 above, I've added three videos and comments for them

**Review by yurytut1993 — APPROVED** [review]
👍

**Review by bc-alexsaiannyi — APPROVED** [review]
looks good

**Review by bc-azvierieva — APPROVED** [review]

**Pull request merged**